### PR TITLE
Feature: Convert PHPUnit tests to Pest

### PR DESCRIPTION
**Pull request by mansoorkhan96**
Hi 👋

This pull request converts PHPUnit tests to Pest. 

Thanks for the amazing work, Team Spatie.

### CODE REVIEW

File: tests/ArrayLogger.php [hunk 1]
```diff
@@ -8,47 +8,47 @@ class ArrayLogger implements LoggerInterface
 {
     protected $logLines = [];
 
-    public function emergency($message, array $context = []): void
+    public function emergency(string|\Stringable $message, array $context = []): void
```
Comment (freekmurze) [inline tests/ArrayLogger.php:11]: Could you import `Stringable`?

File: tests/ImageTest.php [hunk 2]
```diff
@@ -1,43 +1,29 @@
...
-class ImageTest extends TestCase
-{
-    /** @test */
-    public function it_will_throw_an_exception_when_given_a_non_existing_file()
-    {
-        $this->expectException(InvalidArgumentException::class);
-
-        new Image('non existing file');
-    }
+it('will throw an exception when given a non existing file', fn () => new Image('non existing file'))
```
Comment (freekmurze) [inline tests/ImageTest.php:5]: Let's use a normal function instead of a short closure here.

**Review by freekmurze — CHANGES_REQUESTED** [review]
Very nice! I only got a few nitpicks.

File: tests/ArrayLogger.php (hunk 1 above)
Comment (mansoorkhan96) [reply tests/ArrayLogger.php]: Done 👍

**Comment by mansoorkhan96** [comment]
Hey, @freekmurze I have updated the PR.

**Comment by freekmurze** [comment]
It seems like every tests is failing. Please check locally and push a fix.

**Comment by mansoorkhan96** [comment]
Changes description:

- Removed type hinting for `$message` argument in `ArrayLogger.php` for backward compatibility tests.
- Updated expectations in `OptimizerTest` to be in Unix style. (Apologies, I am on windows and did not notice this issue).
- Replaced short closure functions with normal functions for backward compatibility tests.

**Comment by mansoorkhan96** [comment]
Hey, Sir @freekmurze 

Does this require further changes? Please let me know when you have time :)

Thanks.

**Comment by freekmurze** [comment]
Looks good, thanks!

**Pull request merged**